src/array/cuda/gather_mm.cu

```diff
@@ -29,13 +29,27 @@ hipblasStatus_t cublasGemm(
   return HIPBLAS_STATUS_EXECUTION_FAILED;
 }
 
+#ifdef DGL_USE_ROCM
+// TODO: ROCM doesn't define this. Figure out what the actual right way to handle this is.
+#define __syncwarp()
```

Comment:
Some advice here.

https://github.com/ROCm/HIP/issues/3668#issuecomment-2479930563

Reply:
Yeah I figured that's what your define was about, but I'm confused the PyTorch appears to just silently noop with `WARP_SYNC` on AMD. I defined it to `__syncthreads()` for now